src/main/java/com/iexec/worker/docker/ComputationService.java

```diff
@@ -113,170 +159,217 @@ public boolean runNonTeeComputation(TaskDescription taskDescription,
+        // to have the same workflow as TEE.
+        String source = workerConfigService.getTaskIexecOutDir(chainTaskId) + IexecFileHelper.SLASH_COMPUTED_JSON;
+        String target = workerConfigService.getTaskOutputDir(chainTaskId) + IexecFileHelper.SLASH_COMPUTED_JSON;
+        boolean isCopied = FileHelper.copyFile(source, target);
+        if (!isCopied) {
+            log.error("Failed to copy computed.json file to /output [chainTaskId:{}]", chainTaskId);
+            return false;
+        }
+        // encrypt result if needed
+        if (taskDescription.isResultEncryption() && !encryptResult(chainTaskId)) {
+            log.error("Failed to encrypt result [chainTaskId:{}]", chainTaskId);
+            return false;
+        }
+        return true;
+    }
 
-        String stdout = appExecutionResult.getStdout()
-                + teePostComputeExecutionResult.getStdout();
+    private String computeResultDigest(ComputedFile computedFile) {
+        String chainTaskId = computedFile.getTaskId() ;
+        String resultDigest;
+        if (iexecHubService.getTaskDescription(chainTaskId).isCallbackRequested()) {
+            resultDigest = ResultUtils.computeWeb3ResultDigest(computedFile);
+        } else {
+            resultDigest = ResultUtils.computeWeb2ResultDigest(computedFile,
+                    workerConfigService.getTaskOutputDir(chainTaskId));
+        }
+        if (resultDigest.isEmpty()) {
+            log.error("Failed to computeResultDigest (resultDigest empty)[chainTaskId:{}, computedFile:{}]",
+                    chainTaskId, computedFile);
+            return "";
+        }
+        return resultDigest;
+    }
 
+    private boolean encryptResult(String chainTaskId) {
+        String beneficiarySecretFilePath = workerConfigService.getBeneficiarySecretFilePath(chainTaskId);
+        String resultZipFilePath = workerConfigService.getTaskOutputDir(chainTaskId);
+        String taskOutputDir = workerConfigService.getTaskOutputDir(chainTaskId);
+        log.info("Encrypting result zip [resultZipFilePath:{}, beneficiarySecretFilePath:{}]",
+                resultZipFilePath, beneficiarySecretFilePath);
+        encryptFile(taskOutputDir, resultZipFilePath, beneficiarySecretFilePath);
+        String encryptedResultFilePath = workerConfigService.getTaskOutputDir(chainTaskId) + FileHelper.SLASH_IEXEC_OUT + ".zip";
+        if (!new File(encryptedResultFilePath).exists()) {
+            log.error("Encrypted result file not found [chainTaskId:{}, encryptedResultFilePath:{}]",
+                    chainTaskId, encryptedResultFilePath);
+            return false;
+        }
+        // replace result file with the encypted one
+        return FileHelper.replaceFile(resultZipFilePath, encryptedResultFilePath);
+    }
 
-        return resultService.saveResult(chainTaskId, taskDescription, stdout);
+    private void encryptFile(String taskOutputDir, String resultZipFilePath, String publicKeyFilePath) {
```

Comment:
We should use same encryption material than tee-post-compute when we will work on that part.

Reply:
Totally agree. Same thing I just moved code around.